### PR TITLE
gpu_ctx_vk: wait any pending semaphores before executing the update command buffer

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/backends/vk/gpu_ctx_vk.c [hunk 1]
```diff
@@ -1054,7 +1059,7 @@ static int vk_begin_draw(struct gpu_ctx *s, double t)
     s_priv->cur_cmd = s_priv->cmds[s_priv->cur_frame_index];
     VkResult res = ngli_cmd_vk_begin(s_priv->cur_cmd);
     if (res != VK_SUCCESS)
-        return res;
+        return ngli_vk_res2ret(res);
```
Comment (ubitux) [inline libnopegl/src/backends/vk/gpu_ctx_vk.c:1085]: the `ngli_cmd_vk_add_wait_sem` ret code below is also a `VkResult` (you adjusted that but in the wrong commit)

**Review by ubitux — CHANGES_REQUESTED** [review]

File: libnopegl/src/backends/vk/gpu_ctx_vk.c [hunk 2]
```diff
@@ -1032,11 +1037,11 @@ static int vk_end_update(struct gpu_ctx *s, double t)
     VkSemaphore update_finished_sem = s_priv->update_finished_sems[s_priv->cur_frame_index];
     VkResult res = ngli_cmd_vk_add_signal_sem(s_priv->cur_cmd, &update_finished_sem);
     if (res != VK_SUCCESS)
-        return res;
+        return ngli_vk_res2ret(res);
 
     res = ngli_cmd_vk_submit(s_priv->cur_cmd);
     if (res != VK_SUCCESS)
-        return res;
+        return ngli_vk_res2ret(res);
```
[the remark this answers is not in the document]
Comment (mbouron) [reply libnopegl/src/backends/vk/gpu_ctx_vk.c:1067]: Fixed in a separate commit.

File: libnopegl/src/backends/vk/gpu_ctx_vk.c (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/backends/vk/gpu_ctx_vk.c:1085]: Done.

**Review by ubitux — APPROVED** [review]
LGTM thanks

**Pull request merged**